File: app/src/main/java/com/example/xc_m6_java/manager/PrefsManager.java
package com.example.xc_m6_java.manager;

import android.content.Context;
import android.content.SharedPreferences;

public class PrefsManager {

    Context context;
    SharedPreferences sharedPreferences;

    public PrefsManager(Context context) {
        sharedPreferences = context.getSharedPreferences("MyPrefs",Context.MODE_PRIVATE);
    }

    public static PrefsManager prefsManager;

    public static PrefsManager getInstance(Context context) {
        if (prefsManager == null) {
            prefsManager = new PrefsManager(context);
        }
        return prefsManager;
    }

    public void saveData(String key, String value) {
        SharedPreferences.Editor prefsEditor = sharedPreferences.edit();
        prefsEditor.putString(key,value);
        prefsEditor.commit();
    }

    public String loadData(String key) {
        return sharedPreferences.getString(key,"");
    }

    public void remove(String key) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.remove(key);
    }

    public void clearAll() {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.clear();
    }

}

File: app/src/main/java/com/example/xc_m6/manager/PrefsManager.kt
package com.example.xc_m6.manager

import android.content.Context
import android.content.SharedPreferences
import android.widget.Toast
import kotlin.coroutines.coroutineContext

class PrefsManager(context: Context) {

    private val sharedPreferences: SharedPreferences?

    companion object {
        private var prefsManager: PrefsManager? = null
        fun getInstance(context: Context): PrefsManager? {
            if (prefsManager == null) {
                prefsManager = PrefsManager(context)
            }
            return prefsManager
        }
    }

    init {
        sharedPreferences = context.getSharedPreferences("MyPrefs",Context.MODE_PRIVATE)
        // A file that named MyPrefs will open in the locale memory
    }

    fun saveData(key: String?, value: String?,context: Context) {
        val prefsEditor = sharedPreferences!!.edit()
        prefsEditor.putString(key,value)
        prefsEditor.commit()
    }

    fun loadData(key: String?): String? {
        return if (sharedPreferences != null) sharedPreferences.getString(key,"") else ""
    }

    fun removeData(key: String?) {
        val prefsEditor = sharedPreferences!!.edit()
        prefsEditor.remove(key)
    }

    fun clearAll() {
        val prefsEditor = sharedPreferences!!.edit()
        prefsEditor.clear()
    }
}